javaparser-core/src/main/java/com/github/javaparser/ast/stmt/SwitchEntry.java

```diff
@@ -140,6 +146,15 @@ public boolean isSwitchStatementEntry() {
         return type == Type.STATEMENT_GROUP;
     }
 
+    /**
+     * This is required for the ConcreteSyntaxModel for deciding if a "when" token should
+     * be expected or not.
+     */
+    @DerivedProperty
+    public boolean isGuardedEntry() {
```

Comment:
I'm not sure that the GUARDED_ENTRY property is necessary. Can't we just use the GUARD property?